optuna/samplers/nsgaii/_sampler.py

```diff
@@ -45,6 +45,34 @@ class NSGAIISampler(BaseSampler):
     - `A fast and elitist multiobjective genetic algorithm: NSGA-II
       <https://doi.org/10.1109/4235.996017>`__
 
+    .. note::
+        :class:`~optuna.samplers.TPESampler` became much faster in v4.0.0 and supports several
+        features not supported by ``NSGAIISampler`` such as handling of dynamic search
+        space and categorical distance. To use :class:`~optuna.samplers.TPESampler`, you need to
+        explicitly specify the sampler as follows:
+
+        .. testcode::
+
+            import optuna
+
+
+            def objective(trial):
+                x = trial.suggest_float("x", -100, 100)
+                y = trial.suggest_categorical("y", [-1, 0, 1])
+                f1 = x**2 + y
+                f2 = -((x - 2) ** 2 + y)
+                return f1, f2
+
+
+            # We minimize the first objective and maximize the second objective.
+            sampler = optuna.samplers.TPESampler()
+            study = optuna.create_study(directions=["minimize", "maximize"], sampler=sampler)
+            study.optimize(objective, n_trials=100)
```

Comment:
Should we include the code example for multi objective here? My first impression is that it seems a bit redundant given the above code example.

<img width="1124" alt="Screenshot 2024-08-28 10 54 25" src="https://github.com/user-attachments/assets/f5332fc5-8eef-4ab3-9f5f-175412128f9d">

Reply:
I removed the code example and enhanced the passage a bit!